eth/tracers/api_blockResult.go

```diff
@@ -0,0 +1,347 @@
+			failed = err
+			break
+		}
+		// Finalize the state so any modifications are written to the trie
+		// Only delete empty objects if EIP158/161 (a.k.a Spurious Dragon) is in effect
+		env.state.Finalise(vmenv.ChainConfig().IsEIP158(block.Number()))
+	}
+	close(jobs)
+	pend.Wait()
+
+	// If execution failed in between, abort
+	if failed != nil {
+		return nil, failed
+	}
+
+	return api.fillBlockResult(env, block)
+}
+
+func (api *API) getTxResult(env *traceEnv, state *state.StateDB, index int, block *types.Block) error {
+	tx := block.Transactions()[index]
+	msg, _ := tx.AsMessage(env.signer, block.BaseFee())
+	from, _ := types.Sender(env.signer, tx)
+	to := tx.To()
+
+	txctx := &Context{
+		BlockHash: block.TxHash(),
+		TxIndex:   index,
+		TxHash:    tx.Hash(),
+	}
+	// Call Prepare to clear out the statedb access list
+	state.Prepare(txctx.TxHash, txctx.TxIndex)
+
+	sender := &types.AccountWrapper{
+		Address:  from,
+		Nonce:    state.GetNonce(from),
+		Balance:  (*hexutil.Big)(state.GetBalance(from)),
+		CodeHash: state.GetCodeHash(from),
+	}
+	var receiver *types.AccountWrapper
+	if to != nil {
+		receiver = &types.AccountWrapper{
+			Address:  *to,
+			Nonce:    state.GetNonce(*to),
+			Balance:  (*hexutil.Big)(state.GetBalance(*to)),
+			CodeHash: state.GetCodeHash(*to),
+		}
+	}
+	// Run the transaction with tracing enabled.
+	vmenv := vm.NewEVM(env.blockCtx, core.NewEVMTxContext(msg), state, api.backend.ChainConfig(), vm.Config{Debug: true, Tracer: env.tracer, NoBaseFee: true})
+	// Computes the new state by applying the given message.
+	env.rMu.Lock()
+	result, err := core.ApplyMessage(vmenv, msg, new(core.GasPool).AddGas(msg.Gas()))
+	if err != nil {
+		return fmt.Errorf("tracing failed: %w", err)
+	}
+	env.rMu.Unlock()
+	// If the result contains a revert reason, return it.
+	returnVal := result.Return()
+	if len(result.Revert()) > 0 {
+		returnVal = result.Revert()
+	}
+
+	createdAcc := env.tracer.CreatedAccount()
+	var after []*types.AccountWrapper
+	if to == nil {
+		if createdAcc == nil {
+			return errors.New("unexpected tx: address for created contract unavailable")
+		}
+		to = &createdAcc.Address
+	}
+	// collect affected account after tx being applied
+	for _, acc := range []common.Address{from, *to, api.backend.Coinbase()} {
+		after = append(after, &types.AccountWrapper{
+			Address:  acc,
+			Nonce:    state.GetNonce(acc),
+			Balance:  (*hexutil.Big)(state.GetBalance(acc)),
+			CodeHash: state.GetCodeHash(acc),
+		})
+	}
+
+	// merge required proof data
+	proofAccounts := env.tracer.UpdatedAccounts()
+	for addr := range proofAccounts {
+		addrStr := addr.String()
+
+		env.pMu.Lock()
+		_, existed := env.Proofs[addrStr]
+		env.pMu.Unlock()
+		if existed {
+			continue
+		}
+		proof, err := state.GetProof(addr)
+		if err != nil {
+			log.Error("Proof not available", "address", addrStr, "error", err)
+			// but we still mark the proofs map with nil array
+		}
```

Comment:
Why not fail here?

Reply:
Can refer the blockResult's cache api [historical change](https://github.com/scroll-tech/go-ethereum/blob/01a0618b02be619a5d1b1f40420dbcd926e31680/miner/worker.go#L716)!